newsfragments/4567.bugfix.rst

```diff
@@ -0,0 +1 @@
+Ensured all methods in ``setuptools.modified`` raise a consistant ``distutils.errors.DistutilsError`` type -- by :user:`Avasam`
```

Comment:
```suggestion
Ensured methods in ``setuptools.modified`` preferably raise a consistent
``distutils.errors.DistutilsError`` type
(except in the deprecated use case of ``SETUPTOOLS_USE_DISTUTILS=stdlib``)
-- by :user:`Avasam`
```

Reply:
> In the case of the exception (e.g. `SETUPTOOLS_USE_DISTUTILS=stdlib`) these methods would raise `setuptools._distutils.error.DistutilsError` error, right?
> 
> So maybe we have to change the news fragment a bit...

hmm, looking at my tests I think the intention was to be truly the same as `distutils.errors.DistutilsError` at all time, but I see how that won't be the case, unless we start wrapping `_distutils` methods here.

I think that's still an improvement for the "intended usage" (ie: not using deprecated `SETUPTOOLS_USE_DISTUTILS=stdlib`). But I'll need to update my tests to reflect that.

Reply:
I think it is OK to say: "If you need to catch that error, please don't use the deprecated SETUPTOOLS_USE_DISTUTILS=stdlib settings", so I am good with this.